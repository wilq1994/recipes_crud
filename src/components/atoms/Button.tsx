import styled, { css } from "styled-components";

const Button = styled.button<{ small?: boolean; secondary?: boolean }>`
  background: transparent;
  color: var(--primary-color);
  font-weight: 700;
  font-size: 1rem;
  padding: 1em 1.5em;
  margin: 0;
  border: 0;
  outline: 0;
  border-radius: 3px;
  box-shadow: 0 0 0 1px var(--primary-color) inset;
  transition: background 0.3s ease, box-shadow 0.3s ease;
  cursor: pointer;
  min-width: 120px;
  &:hover {
    background: #fff;
    box-shadow: 0 0 0 3px var(--primary-color) inset;
  }

  ${(props) =>
    props.small &&
    css`
      padding: 0.5em 1em;
      min-width: auto;
    `}

  ${(props) =>
    props.secondary &&
    css`
        color: var(--tertiary-color);
        box-shadow: 0 0 0 1px var(--tertiary-color) inset;
        &:hover {
            box-shadow: 0 0 0 3px var(--tertiary-color) inset;
        }
    `}
`;

export default Button;
