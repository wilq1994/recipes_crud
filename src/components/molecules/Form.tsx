import React, { ChangeEvent, Dispatch } from "react";

import { ActionTypes, RecipeFormType } from "components/types";
import { Action } from "components/reducer";
import FormField from "components/atoms/FormField";

type Props = {
  form: RecipeFormType;
  dispatch: Dispatch<Action>;
};

const Form = ({ form, dispatch }: Props) => {
  function onNameChange(event: ChangeEvent<HTMLInputElement>) {
    dispatch({ type: ActionTypes.SET_NAME, payload: event?.target.value });
  }

  function onIngredientsChange(event: ChangeEvent<HTMLTextAreaElement>) {
    dispatch({
      type: ActionTypes.SET_INGREDIENTS,
      payload: event?.target.value,
    });
  }

  return (
    <div>
      <FormField>
        <label>Recipe</label>
        <input type="text" value={form.name} onChange={onNameChange} />
      </FormField>
      <FormField>
        <label>Ingredients</label>
        <textarea value={form.ingredients} onChange={onIngredientsChange} />
      </FormField>
    </div>
  );
};

export default Form;
