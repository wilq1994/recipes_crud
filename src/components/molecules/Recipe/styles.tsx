import styled from "styled-components";

import Button from "components/atoms/Button";

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;

  h2 {
    color: var(--dark-blue-color);
    text-transform: uppercase;
    font-size: 0.75rem;
    font-weight: 400;
    font-weight: 700;
    letter-spacing: 1px;
  }
`;

export const Buttons = styled.div`
  display: flex;
  justify-content: flex-end;
  ${Button} + ${Button} {
    margin-left: 0.7rem;
  }
`;
