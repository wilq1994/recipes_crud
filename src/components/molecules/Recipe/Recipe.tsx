import React from "react";

import List from "components/atoms/List";
import Button from "components/atoms/Button";

import { Header, Buttons } from "./styles";

type Props = {
  ingredients: string[];
  onDelete: () => void;
  onEdit: () => void;
};

export const Recipe = ({ ingredients, onDelete, onEdit }: Props) => {
  return (
    <>
      <Header>
        <h2>Ingredients</h2>
        <Buttons>
          <Button onClick={onDelete} small secondary>
            Delete
          </Button>
          <Button onClick={onEdit} small>
            Edit
          </Button>
        </Buttons>
      </Header>
      <List list={ingredients} />
    </>
  );
};
