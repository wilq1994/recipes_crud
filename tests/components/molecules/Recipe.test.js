import React from "react";
import { shallow } from "enzyme";

import {Recipe} from "components/molecules/Recipe";
import Button from "components/atoms/Button";

describe("Recipe", () => {
  it("should render and match snapshot", () => {
    const fakeOnDelete = jest.fn();
    const fakeOnEdit = jest.fn();
    const component = shallow(
        <Recipe
            ingredients={['a','b','c']}
            onDelete={fakeOnDelete}
            onEdit={fakeOnEdit}
        />
    );

    expect(component.exists()).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it("should call onDelete", () => {
    const fakeOnDelete = jest.fn();
    const fakeOnEdit = jest.fn();
    const component = shallow(
        <Recipe
            ingredients={['a','b','c']}
            onDelete={fakeOnDelete}
            onEdit={fakeOnEdit}
        />
    );

    component.find(Button).at(0).simulate('click')
    expect(fakeOnDelete).toHaveBeenCalled()
  });

  it("should call onEdit", () => {
    const fakeOnDelete = jest.fn();
    const fakeOnEdit = jest.fn();
    const component = shallow(
        <Recipe
            ingredients={['a','b','c']}
            onDelete={fakeOnDelete}
            onEdit={fakeOnEdit}
        />
    );

    component.find(Button).at(1).simulate('click')
    expect(fakeOnEdit).toHaveBeenCalled()
  });
});
