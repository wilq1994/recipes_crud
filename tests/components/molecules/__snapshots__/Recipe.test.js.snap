// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should render and match snapshot 1`] = `
<Fragment>
  <styled.header>
    <h2>
      Ingredients
    </h2>
    <styled.div>
      <styled.button
        onClick={[MockFunction]}
        secondary={true}
        small={true}
      >
        Delete
      </styled.button>
      <styled.button
        onClick={[MockFunction]}
        small={true}
      >
        Edit
      </styled.button>
    </styled.div>
  </styled.header>
  <List
    list={
      Array [
        "a",
        "b",
        "c",
      ]
    }
  />
</Fragment>
`;
